### PR TITLE
Improve linter

### LINKED ISSUE #160: Add lint rules
* make errors for unused imports and exports
* prevent circular dependencies

### PR DESCRIPTION
**Pull request by giladfuchs**
# Description
this to resolve #160 issue
clean all unused code with ts-prune

and remove unused export.

add ci in lint check for circular dependencies

add for linter not allowed unused code like
    "@typescript-eslint/no-unused-vars": "error",
that was mostly react import so I also add this
    "react/react-in-jsx-scope": "off",

### CODE REVIEW

**Comment by giladfuchs** [comment]
after falling 
`yarn build`

since there is unused import in node modules I put it back to normal
for passing the test.

File: tsconfig.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "allowJs": false,
     "skipLibCheck": true,
     "esModuleInterop": true,
+    "noUnusedLocals": false,
```
Comment (NoamGaash) [inline tsconfig.json:12]: why?

**Review by NoamGaash — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 2]
```diff
@@ -23,6 +23,8 @@
   "plugins": ["react", "@typescript-eslint"],
   "rules": {
     "@typescript-eslint/no-non-null-assertion": "off",
+    "@typescript-eslint/no-unused-vars": "off",
+    "react/react-in-jsx-scope": "off",
```
[the remark this answers is not in the document]
Comment (giladfuchs) [reply .eslintrc.json:27]: we need to fix 
"open-bus-stride-client": "0.1.2",

can I do in this repository  pull request 
if yes plz give me link to the repo cause I don't find it

File: tsconfig.json (hunk 1 above)
Comment (giladfuchs) [reply tsconfig.json:12]: when you change it to true it raise an error on unused variables

Comment (giladfuchs) [reply tsconfig.json]: ok so i remove it.
stay only with linter.
then node modules not make a problem

File: .eslintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (giladfuchs) [reply .eslintrc.json]: changed to "error"

File: .eslintrc.json [hunk 3]
```diff
@@ -20,9 +20,12 @@
     "ecmaVersion": "latest",
     "sourceType": "module"
   },
+
   "plugins": ["react", "@typescript-eslint"],
   "rules": {
     "@typescript-eslint/no-non-null-assertion": "off",
+    "@typescript-eslint/no-unused-vars": "error",
+    "react/react-in-jsx-scope": "off",
```
Comment (NoamGaash) [inline .eslintrc.json:28]: is it needed?

Comment (NoamGaash) [reply .eslintrc.json:28]: I learned something new
https://dev.to/titungdup/you-no-longer-need-to-import-react-from-react-3pbj

**Pull request merged**